fhir-config/src/main/java/com/ibm/fhir/config/FHIRConfigHelper.java

```diff
@@ -74,9 +74,16 @@ private static JsonValue getPropertyFromTenantOrDefault(String propertyName) {
         // let's try to find it in the default config.
         if (result == null && !FHIRConfiguration.DEFAULT_TENANT_ID.equals(tenantId)) {
             try {
-                pg = FHIRConfiguration.getInstance().loadConfiguration();
-                if (pg != null) {
-                    result = pg.getJsonValue(propertyName);
+                if (propertyName.startsWith(FHIRConfiguration.PROPERTY_DATASOURCES)) {
+                    // Issue #639. Prevent datasource lookups from falling back to 
+                    // the default datasource which breaks tenant isolation.
+                    result = null;
```

Comment:
I don't think this is necessary.  line 75 checks... `result==null`
perhaps just make this an if without an else branch?

Reply:
Correct, this was just a style thing so that I could add that comment in a clear way rather than having to deal with negatives.

Reply:
Good - it was intentional.  LGTM